shared/src/test/scala/org/adridadou/openlaw/vm/OpenlawVmSpec.scala

```diff
@@ -87,6 +87,40 @@ class OpenlawVmSpec extends FlatSpec with Matchers {
     vm1.signature(identity2.email) shouldBe None
   }
 
+  it should "be possible to sign the contract with an external signature" in {
+    val template = """this is a contract [[Signatory:ExternalSignature(service:"DocuSign")]]"""
+    val Right(identity: Identity) = Email("hello@world.com").map(Identity.withEmail)
+    val templateId = TemplateId(TestCryptoService.sha256(template))
+
+    val serviceAccount = TestAccount.newRandom
+    val serviceName = ServiceName("DocuSign")
+
+    val definition = ContractDefinition(
+      creatorId = UserId.SYSTEM_ID,
+      mainTemplate = templateId,
+      templates = Map(),
+      parameters = TemplateParameters(Map(VariableName("Signatory") -> ExternalSignatureType.internalFormat(ExternalSignature(serviceName = serviceName, identity = Some(identity))).right.value))
```

Comment:
I noticed you passed the value as template parameter, but wasn't sure if we need that declared.